### ISSUE TITLE
DM derivative output format from TEMPO not recognized by PINT

### ISSUE THREAD

**Issue by scottransom** [nanograv/PINT#1188]
While PINT can handle DM time derivatives for fitting, the format of the parameters as output by TEMPO (i.e. `DM001`, `DM002`, ...) is not recognized.  PINT expects and ouputs `DM1`, `DM2`, etc.  That fomat *is* accepted by TEMPO as input.  So we just need to improve the parameter parsing (for prefix parameters, I suppose) to recognize the format with the preceeding zeros.

Here is an error example:
```
/home/sransom/python_venvs/py39/lib/python3.9/site-packages/pint/models/model_builder.py:114: UserWarning: Unrecognized parfile line 'DM001 -1.694891583439D-04 1 2.283525222389D-04'
  warnings.warn(f"Unrecognized parfile line '{p_line}'", UserWarning)
/home/sransom/python_venvs/py39/lib/python3.9/site-packages/pint/models/model_builder.py:114: UserWarning: Unrecognized parfile line 'DM002 1.246760990803D-03 1 4.987934997885D-05'
  warnings.warn(f"Unrecognized parfile line '{p_line}'", UserWarning)
```

**Comment by tcromartie** [comment]
Hey @scottransom, I'm trying to figure out the best way to do this. Should it be a change to the way prefixes work (i.e. always ignore the zeros)? I can't think of another parameter that would affect, but I'm not sure. Should it instead be a hardcoded change to make DM00n an alias for DMn? Should it affect both reading and writing? I was sort of thinking this would be as easy as adding an alias somewhere but that doesn't seem to be the case.

**Comment by tcromartie** [comment]
Yeah I'm confused about that too, I thought the second splitting case would take care of it

**Comment by scottransom** [comment]
I think the issue might be because `DM1` is explicitly defined in `dispersion_model.py` without an alias to `DM001`.  If you look in `parameter.py`, you can define aliases to a parameter.  So maybe just try putting that in?  And then maybe the other ones which are not already defined will just work?

**Comment by scottransom** [comment]
Hmmm.  That didn't seem to help...  It is some subtle thing deep in there.  It will probably be a trivial change once we figure out where it goes!

**Comment by scottransom** [comment]
Next place to look is the loop starting on line 337 in `model_builder.py`.  I suspect we aren't treating something correctly there.

**Comment by scottransom** [comment]
Ah-ha...  OK.  This should help.
```
In [3]: ac = AllComponents()

In [4]: ac.alias_to_pint_param("DM1")
Out[4]: ('DM1', 'DM1')

In [5]: ac.alias_to_pint_param("DM001")
---------------------------------------------------------------------------
UnknownParameter                          Traceback (most recent call last)
<ipython-input-5-088e5d939cb7> in <module>
----> 1 ac.alias_to_pint_param("DM001")

~/git/PINT/src/pint/models/timing_model.py in alias_to_pint_param(self, alias)
   3039             first_init_par = pint_par
   3040         if pint_par is None:
-> 3041             raise UnknownParameter(
   3042                 "Can not find matching PINT parameter for '{}'".format(alias)
   3043             )

UnknownParameter: Can not find matching PINT parameter for 'DM001'
```

**Comment by tcromartie** [comment]
Ok the problem can obviously be fixed by adding a special case to parse_parfile, but should this happen on a less-deep level (like changing the parfile_dict later, or something else...), and should it apply to all params in "known param + leading zero(s) + index" format, or just be a special exception for DM00n? Sorry for asking so many questions, this is just an area where I need to learn what the preferred "PINT approach" to things is.

**Comment by aarchiba** [comment]
I think it would be much better to strip the leading zeros - don't we already accept DMX0001 and other similar things? I agree it's not clear why the three-digit version doesn't work, but tracking that down would be a much nicer solution.

**Comment by tcromartie** [comment]
Addressed by #1204. Can still talk about how the formats (pint/tempo(2)) change how parameters like these are written out, but this bug should be fixed.
